pkg/rest/config_upload.go

```diff
@@ -188,13 +188,21 @@ func getObjectIdIfAlreadyExists(client *http.Client, api api.Api, url string, ob
 		return "", err
 	}
 
+	var found = false
+	var configName = ""
 	for i := 0; i < len(values); i++ {
 		value := values[i]
 		if value.Name == objectName {
-			return value.Id, nil
+			if found {
+				util.Log.Warn("\t\t\tFound two or more configs having same name: %s. Please delete duplicates.", objectName)
+				return configName, nil
+			} else {
+				found = true
+				configName = value.Id
+			}
 		}
 	}
-	return "", nil
+	return configName, nil
```

Comment:
did you notice any significant performance impact of now iterating through the complete values instead of exiting on the first match?

Reply:
No noticeable difference on cluster config, loading payload takes the most time but nothing changed there.